backend/src/LibExecution/ProgramTypesToRuntimeTypes.fs

```diff
@@ -206,11 +206,16 @@ module Expr =
           match op with
           | PT.BinOpAnd -> RT.EAnd(id, prev, toRT expr)
           | PT.BinOpOr -> RT.EOr(id, prev, toRT expr)
-        | PT.EPipeEnum (id, typeName, caseName, _) ->
-          RT.EEnum(id, FQTypeName.toRT typeName, caseName, [ prev ])
+        | PT.EPipeEnum (id, typeName, caseName, fields) ->
+          let fields' =
+            fields
+            // CLEANUP: We probably shouldn't put an EUnit type argument into EEnum at the parser level.
+            |> List.filter (function
```

Comment:
I think this is an artifact of the parser or maybe just the tests.

For the test `(3 |> Ok()) = Ok 3`, we should actually write this test as `(3 |> Ok) = Ok 3`. I think that doesn't work out of the box though, and I suspect you need to update the parser to make it work.

The parser is a little finicky, let me know if you'd like me to take a shot at it.